### PR TITLE
compositor: Drop vestigial queueing support

### PR DESCRIPTION
**Pull request by RAOF**
We no longer need to be able to set a per-surface SwapInterval; that's handled client side. Likewise, we don't need to support submitting a queue of buffers to a surface; Wayland specifies mailbox¹ semantics.

Drop some of the infrastructure needed to support these features we no longer use anywhere (including QtMir)

¹: ie: framedropping

### CODE REVIEW

**Review by AlanGriffiths — CHANGES_REQUESTED** [review]
Looks sensible. It seems a shame to lose the flexibility, but we don't need it now.

File: src/server/compositor/multi_monitor_arbiter.h [hunk 1]
```diff
@@ -36,15 +35,14 @@ class Schedule;
...
+    MultiMonitorArbiter();
     ~MultiMonitorArbiter();
 
     std::shared_ptr<graphics::Buffer> compositor_acquire(compositor::CompositorID id) override;
     std::shared_ptr<graphics::Buffer> snapshot_acquire() override;
-    void set_schedule(std::shared_ptr<Schedule> const& schedule);
     bool buffer_ready_for(compositor::CompositorID id);
-    void advance_schedule();
+
+    void submit_buffer(std::shared_ptr<graphics::Buffer> buffer);
```
[the remark this answers is not in the document]
Comment (RAOF) [reply src/server/compositor/multi_monitor_arbiter.h:45]: Hm. I thought our resolution on that was:
* Pass `std::shared_ptr` (and `std::unique_ptr`) by value if ownership is definitely going to be taken
* Pass `std::shared_ptr<T>` by `const` reference if ownership is conditionally taken
* Pass by `T&` if ownership if definitely not taken.

I don't have particularly strong opinions, other than that `std::shared_ptr<T>&&` is weird.

[the remark this answers is not in the document]
Comment (AlanGriffiths) [reply src/server/compositor/multi_monitor_arbiter.h:45]: Anyway, this is moot. We should be doing a `shared_ptr::swap()` in the implementation

**Review by AlanGriffiths — APPROVED** [review]
LGTM now

**Pull request merged**